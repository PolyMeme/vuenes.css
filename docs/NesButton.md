# nes-button 

Vue Component implementation of the Nes Button 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## slots 

- `default` Use this slot for the content Inside the button 

## props 

- `primary` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Primary. Will use the `is-primary` class 

- `success` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Success. Will use the `is-success` class 

- `warning` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Warning. Will use the `is-warning` class 

- `error` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Error. Will use the `is-error` class 

- `disabled` ***Boolean*** (*optional*) 

  Prop that define if the Button is type Disabled. Will use the `is-disabled` class 

## computed properties 

- `buttonClasses` 

  Computed Property that uses the button classes depending of the props used. 

   **dependencies:** `primary`, `success`, `warning`, `error`, `disabled` 


## events 

- `click` 

  Fired when a the user clicks the button. 

## methods 

- `click()` 

  Method that emits the click event 

