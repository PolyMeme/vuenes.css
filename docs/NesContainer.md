# nes-container 

Vue Component implementation of the Nes Container 

- **author** - Jesus Fernando Alvarez Franco 
- **license** - MIT 

## slots 

- `default` Slot used for the Container Content 

## props 

- `title` ***String*** (*optional*) `default: Empty String` 

  Defines the title of the container 

- `center` ***Boolean*** (*optional*) `default: false` 

  Define if the container is centered. Will use the `is-centered` class 

- `dark` ***Boolean*** (*optional*) `default: false` 

  Define if the container is dark. Will use the `is-dark` class 

- `form` ***Boolean*** (*optional*) `default: false` 

  Define if the container is a form. Will use the `form` class 

- `balloon` ***Boolean*** (*optional*) `default: false` 

  Define if the container is for balloons. Will use the `balloon` class 

- `rounded` ***Boolean*** (*optional*) `default: false` 

  Define if the container has rounded corners. Will use the `is-rounded` class 

## computed properties 

- `containerClasses` 

  Computed Property that uses the container classes depending of the props used. 

   **dependencies:** `title`, `center`, `dark`, `rounded`, `form`, `balloon` 


