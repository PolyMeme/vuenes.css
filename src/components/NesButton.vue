<template>
  <button
    class="nes-btn"
    :class="buttonClasses"
    type="button"
    @click="click"
  >
    <!-- Use this slot for the content Inside the button -->
    <slot />
  </button>
</template>

<script>
/**
 * Vue Component implementation of the Nes Button
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {
  props: {
    /**
     * Prop that define if the Button is type Primary. Will use the `is-primary` class
     * @default false
     */
    primary: Boolean,
    /**
     * Prop that define if the Button is type Success. Will use the `is-success` class
     * @default false
     */
    success: Boolean,
    /**
     * Prop that define if the Button is type Warning. Will use the `is-warning` class
     * @default false
     */
    warning: Boolean,
    /**
     * Prop that define if the Button is type Error. Will use the `is-error` class
     * @default false
     */
    error: Boolean,
    /**
     * Prop that define if the Button is type Disabled. Will use the `is-disabled` class and will not trigger the click event
     * @default false
     */
    disabled: Boolean
  },
  computed: {
    /**
     * Computed Property that uses the button classes depending of the props used.
     * @returns {{'is-primary': BooleanConstructor | *, 'is-error': BooleanConstructor | *, 'is-success': BooleanConstructor | *, 'is-disabled': BooleanConstructor | *, 'is-warning': BooleanConstructor | *}}
     */
    buttonClasses () {
      return {
        'is-primary': this.primary,
        'is-success': this.success,
        'is-warning': this.warning,
        'is-error': this.error,
        'is-disabled': this.disabled
      }
    }
  },
  methods: {
    /**
     * Method that emits the click event
     */
    click () {
      /**
       * Fired when a the user clicks the button and the button is not disabled
       */
      if (!this.disabled) {
        this.$emit('click')
      }
    }
  }
}
</script>
