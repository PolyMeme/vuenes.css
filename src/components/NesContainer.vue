<template>
  <section
    :class="containerClasses"
    class="nes-container"
  >
    <h2
      v-if="title"
      class="title"
    >
      {{ title }}
    </h2>
    <!-- Slot used for the Container Content -->
    <slot />
  </section>
</template>

<script>
/**
 * Vue Component implementation of the Nes Container
 * @author Jesus Fernando Alvarez Franco
 * @license MIT
 */
export default {

  props: {
    title: {
      type: String,
      default: () => ''
    },
    center: Boolean,
    dark: Boolean,
    form: Boolean,
    balloon: Boolean
  },
  computed: {
    containerClasses () {
      return {
        'with-title': !!this.title,
        'is-centered': this.center,
        'is-dark': this.dark,
        'form': this.form,
        'balloon': this.balloon
      }
    }
  }
}
</script>

<style lang="scss">
@import "~typeface-press-start-2p/index.css";
@import "~nes.css/scss/base/index";
@import "~nes.css/scss/elements/containers";
</style>
