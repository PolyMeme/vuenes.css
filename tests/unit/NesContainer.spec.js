import { mount } from '@vue/test-utils'
import NesContainer from '../../src/components/NesContainer.vue'
import NesButton from '../../src/components/NesButton.vue'

describe('NesContainer.vue', () => {
  const createContainerWrapper = (propsData = {}, slot) => {
    return mount(NesContainer, {
      propsData,
      slots: {
        default: slot || 'This is a container!'
      }
    })
  }

  it('should render component properly without props', () => {
    const wrapper = createContainerWrapper()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.find('div.container').exists()).toBe(true)
    expect(wrapper.text()).toBe('This is a container!')
    expect(wrapper.find('h2.title').exists()).toBe(false)
  })

  it('should render container with "section" html tag', () => {
    const wrapper = createContainerWrapper({
      tag: 'section'
    })

    expect(wrapper.find('section.container').exists()).toBe(true)
  })

  it('should render container title with default html tag', () => {
    const wrapper = createContainerWrapper({
      title: 'Castlevania: Rondo of Blood'
    })

    const titleWrapper = wrapper.find('div.container.with-title>h2.title')
    expect(titleWrapper.exists()).toBe(true)
    expect(titleWrapper.html().trim()).toBe('<h2 class="title">\n    Castlevania: Rondo of Blood\n  </h2>')
    expect(titleWrapper.text()).toBe('Castlevania: Rondo of Blood')
  })

  it('should render container title with h1 html tag', () => {
    const wrapper = createContainerWrapper({
      titleTag: 'h1',
      title: 'Castlevania: Rondo of Blood'
    })

    const titleWrapper = wrapper.find('div.container.with-title>h1.title')
    expect(titleWrapper.exists()).toBe(true)
    expect(titleWrapper.html().trim()).toBe('<h1 class="title">\n    Castlevania: Rondo of Blood\n  </h1>')
    expect(titleWrapper.text()).toBe('Castlevania: Rondo of Blood')
  })

  it('should set "is-center" class if the center prop is set', () => {
    const wrapper = createContainerWrapper({
      center: true
    })

    const centerContainer = wrapper.find('div.container.is-center')

    expect(wrapper.classes()).toContainEqual('is-center')
    expect(centerContainer.exists()).toBe(true)
  })

  it('should set "is-dark" class if the dark prop is set', () => {
    const wrapper = createContainerWrapper({
      dark: true
    })

    const darkContainer = wrapper.find('div.container.is-dark')

    expect(wrapper.classes()).toContainEqual('is-dark')
    expect(darkContainer.exists()).toBe(true)
  })

  it('should set "is-form" class if the form prop is set', () => {
    const wrapper = createContainerWrapper({
      form: true
    })

    const formContainer = wrapper.find('div.container.form')

    expect(wrapper.classes()).toContainEqual('form')
    expect(formContainer.exists()).toBe(true)
  })

  it('should render slot', () => {
    const wrapper = createContainerWrapper({
      tag: 'article',
      title: 'This is my title',
      titleTag: 'h3',
      center: true,
      dark: true
    }, NesButton)

    expect(wrapper.find('article.container.with-title.is-center.is-dark>h3.title+button[type="button"].btn').exists()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = createContainerWrapper({
      tag: 'article',
      title: 'This is my title',
      titleTag: 'h3',
      center: true,
      dark: true
    }, NesButton)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
