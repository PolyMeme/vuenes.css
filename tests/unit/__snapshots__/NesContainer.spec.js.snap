// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NesContainer.vue should match snapshot 1`] = `
<article class="container with-title is-center is-dark">
  <h3 class="title">
    This is my title
  </h3> <button type="button" class="btn"></button>
</article>
`;
